crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -146,6 +147,56 @@ impl Format<PyFormatContext<'_>> for FormatDanglingComments<'_> {
     }
 }
 
+/// Formats the content of the passed comment.
+///
+/// * Adds a whitespace between `#` and the comment text except if the first character is a `#`, `:`, `'`, or `!`
+/// * Replaces non breaking whitespaces with regular whitespaces except if in front of a `types:` comment
+const fn format_comment(comment: &SourceComment) -> FormatComment {
+    FormatComment { comment }
+}
+
+struct FormatComment<'a> {
+    comment: &'a SourceComment,
+}
+
+impl Format<PyFormatContext<'_>> for FormatComment<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let slice = self.comment.slice();
+        let comment_text = slice.text(SourceCode::new(f.context().contents()));
+
+        let (content, mut start_offset) = comment_text
+            .strip_prefix('#')
+            .map_or((comment_text, TextSize::new(0)), |rest| {
+                (rest, '#'.text_len())
+            });
+
+        write!(f, [source_position(slice.start()), text("#")])?;
```

Comment:
would it make sense to check if the comment looks fine (which should be the great majority of comments) and if so, write it back in a single write?